src/main/java/performance/LocustOperations.java

```diff
@@ -64,37 +67,76 @@ public void executeTask(DataTable data) throws Exception {
 	 * This method raise the master with the parameters of the test defined in cucumber
 	 */
     public void executeMaster() {
-
+        Process process;
         String OPERATING_SYSTEM = System.getProperty("os.name").toLowerCase();
         String command="-f "+ masterFilePath +" --master --no-web --csv="+csvReportFilePath +"/"+ NAMEOFREPORT +" --expect-slaves=1 -c "+ maxUsers +" -r "+ usersLoadPerSecond+" -t"+testTime+"m";
 
         if (OPERATING_SYSTEM.indexOf("win") >= 0) {
-
-        command = "cmd.exe /c start /MIN locust.exe " + command;
-           }
-        else {
-        command= "locust " + command;
+        	command = "cmd.exe /c start /MIN locust.exe " + command;
+        } else {
+        	command= "locust " + command;
         }
-          try {
+        try {
         	process = Runtime.getRuntime().exec(command);
         } catch (IOException error) {
-           System.out.println(error.getMessage());
+        	logger.error("Something went wrong executing the master");
         }
     }
     
+    //This method clear all the values;
+	private void clearValues() {
+		this.locustTask = "";
+		this.maxUsers = 0;
+		this.usersLoadPerSecond = 0;
+		this.maxRPS = 0;
+		this.weight = 0;
+		this.testTime=0;
+	}
+    
     /*
      * This method execute the sequence needed for run the test
      */
     public void executePerformanceTask(DataTable testData) throws Exception {
-        setTestData(testData);
-        executeMaster();
-        TimeUnit.SECONDS.sleep(5);
-        setUpSlave();
-        executeTask(testData);
+        this.setTestData(testData);
+        this.executeMaster();
+        TimeUnit.SECONDS.sleep(10);
+        this.setUpSlave();
+        this.executeTask(testData);
         TimeUnit.MINUTES.sleep(this.testTime);
-        locust.stop();
+        this.locust.stop();
+        this.clearValues();
     }
+    
+    //It returns true or false if the Max response time is higher or not than the expected 
+    public Boolean checkMaxResponseTime(DataTable testData) {
+    	Boolean higher = false;
+    	List<String[]> data = FileOperations.getInstance().readCSV(FileOperations.getInstance().getAbsolutePath(ConfigReader.getInstance().getRequestReportPath()));    	
+    	try {
+			if (this.getMaxResponseTime(data, (data.size()-1))>Long.parseLong(AuxiliarMethods.getInstance().getDataTableValue(testData, "Expected Time"))){
+				higher = true;
+			}
+		} catch (Exception e) {
+			logger.error("Something went wrong cheking the MaxResponseTime");
+		};
+    	return higher;
+    }  
 	
+    //It returns the Max Response Time of the report
+    private Long getMaxResponseTime(List<String[]> data, int testResultsIteration) throws Exception {
+    	int position = 555;
```

Comment:
Why 555?

Reply:
Could be whatever, is just an initialization. 555 to ensure that is not one that could be valid.

Reply:
Can you intialize with null? any other number would be confusing

Reply:
An int can't be null.